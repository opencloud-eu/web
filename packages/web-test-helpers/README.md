# web-test-helpers

This packages provides utilities for unit testing within the OpenCloud app ecosystem.

## Installation

Depending on your package manager, run one of the following commands:

```
$ npm install @opencloud-eu/web-test-helpers --save-dev

$ pnpm add -D @opencloud-eu/web-test-helpers

$ yarn add @opencloud-eu/web-test-helpers --dev
```

Make sure that you have `vue`, `@vue/test-utils`, and `@opencloud-eu/web-pkg` as dev dependencies of your app since those are peer dependencies of this package.
