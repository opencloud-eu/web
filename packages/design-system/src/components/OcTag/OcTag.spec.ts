import { shallowMount } from '@opencloud-eu/web-test-helpers'
import Tag from './OcTag.vue'

describe('OcTag', () => {
  it('uses correct component when type is specified', () => {
    const wrapper = shallowMount(Tag, {
      props: {
        type: 'button'
      }
    })

    expect(wrapper.element.tagName.toLowerCase()).toMatch('button')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('emits a click event', async () => {
    const wrapper = shallowMount(Tag, {
      props: {
        type: 'a'
      }
    })

    await wrapper.trigger('click')
    expect(wrapper.emitted().click).toBeTruthy()
  })
})
