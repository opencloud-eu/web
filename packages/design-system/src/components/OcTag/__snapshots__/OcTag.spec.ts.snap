// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OcTag > uses correct component when type is specified 1`] = `
"<button class="oc-tag oc-tag-m oc-tag-button oc-tag-color-secondary oc-tag-appearance-outline" to="">
  <!-- @slot Content of the tag -->
</button>"
`;
